libraries/cmd/unit_generic.cpp

```diff
@@ -1397,11 +1397,21 @@ void Unit::ProcessDeleteQueue() {
 #ifdef DESTRUCTDEBUG
         if ( unit_delete_queue.back()->isSubUnit() ) {
             VS_LOG(debug, "Subunit Deleting (related to double dipping)");
+            unit_delete_queue.pop_back();
+            continue;
         }
 #endif
         Unit *mydeleter = unit_delete_queue.back();
         unit_delete_queue.pop_back();
+
+        // Avoid segfault when the unit getting destroyed is the player's current target
+        Unit* parent = _Universe->AccessCockpit()->GetParent();
+        if (parent && parent->Target() == mydeleter) {
+            parent->SetTarget(nullptr);
+        }
```

Comment:
Here, to avoid segfaults, we simply traverse the object hierarchy starting with `_Universe->AccessCockpit()->GetParent()`, which returns the player's ship. This way, we don't have to add new parameters all over the place.